server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalAutoDateHistogram.java

```diff
@@ -234,13 +239,19 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeNamedWriteable(format);
         out.writeList(buckets);
         out.writeVInt(targetBuckets);
+        if (out.getVersion().onOrAfter(Version.V_8_2_0)) {
+            out.writeVLong(bucketInnerInterval);
+        }
     }
 
-    public DateHistogramInterval getInterval() {
+    long getBucketInnerInterval() {
+        return bucketInnerInterval;
+    }
 
+    public DateHistogramInterval getInterval() {
         RoundingInfo roundingInfo = this.bucketInfo.roundingInfos[this.bucketInfo.roundingIdx];
         String unitAbbreviation = roundingInfo.unitAbbreviation;
-        return new DateHistogramInterval(Long.toString(bucketInnerInterval) + unitAbbreviation);
+        return new DateHistogramInterval(bucketInnerInterval + unitAbbreviation);
```

Comment:
I'm nitpicking, but I think explicitly calling `Long.toString` is more readable.

Reply:
my IDE told me to remove it, so did I :)

Reply:
Java: "Operator overloading is terrible because C++ overloaded bit shift to mean print!"
Also Java: "`long` + `String` means concatenate, lol"

At any rate, I don't care enough to argue about it.